template/.github/workflows/release.yml

```diff
@@ -5,6 +5,10 @@ on:
     branches:
       - main
 
+permissions:
+  contents: write
+  pull-requests: write
+
```

Comment:
Can you link to where this is explained in the release-please-action repository? I have not encountered the need for this configuration and cannot see any changes documented.

Reply:
I've raised it in my PR ☝🏻 .

It happens when: 

- You're using the default github token and not a PAT
- You haven't enabled the "omg let the bot do everything i don't care about security" settings on your account/org/repo.

So because you're not using a PAT (desirable under as many scenarios as you can get away with), you need to declare to github what the github bot is allowed to do.

release please makes a number of api calls to create : 

- labels
- comments
- PRs


These allare protected by githubs permission system, most people go way too far and use a PAT.

I like to instead try to lean as much as possible on the github token, I find there are very few scenarios where you need a PAT.

### Where do the `release-please` team tal about permissions? 

sadly not in the obvious place.

But they mention it in obscure places, like: 

- https://github.com/googleapis/release-please/pull/1399
- https://github.com/googleapis/release-please/pull/894
- again more about labels: https://github.com/googleapis/release-please/pull/720#discussion_r568058767
- https://github.com/googleapis/release-please/pull/643


I mean I could go on and on, they know there are permissions required but don't mention it? `o.O`

I understand why most action authors don't realise they need to document this: 

- they develop with a PAT, so this problem is hidden from them during development
- github themselves don't document that using a particular api requires a particular permission (they just say that [permissions exist yo](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs) and that "[lulz you should use permission](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#enabling-workflows-for-private-repository-forks)s"! but they never go into [detail about what is affected](https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#create-a-pull-request))

Reply:
Additionally, here is my first run on a fresh repo using the default workflow provided by the template: 

1. results in a fail: not allowed to access resource (classic no permission error)
[![image](https://user-images.githubusercontent.com/61225/221415593-8815a381-6273-43aa-b632-c70bc5db25a7.png)](https://github.com/airtonix/asdf-xc/actions/runs/4258843119/jobs/7410484748)

2. moving along to the next significant change (i add `contents:write`) which isn't enough, because it then failed on creating the PR:
[![image](https://user-images.githubusercontent.com/61225/221415916-86440057-099b-4bba-afdb-115760d32bab.png)](https://github.com/airtonix/asdf-xc/actions/runs/4267297782/jobs/7428743629)

3. finally I add `pull_requests: write` and also make changes in my settings: 
[![image](https://user-images.githubusercontent.com/61225/221415869-ad41bf26-7d37-4e8e-88d2-5d66c3bfeb8a.png)](https://github.com/airtonix/asdf-xc/actions/runs/4267348844/jobs/7428855533)

Reply:
actually I lied (apparently), github hides all the permissions required by their api here: 

https://docs.github.com/en/rest/overview/permissions-required-for-github-apps

They have a lot of permissions, that only a small subset are used by github actions.

Reply:
@joshukraine You've probably allowed github actions bot to read and write on all scopes which is why you've not experienced this: 
![image](https://user-images.githubusercontent.com/61225/221416080-f513e506-023d-40ac-9886-0891faaf9ea3.png)

Reply:
in fact, you probably don' even need to tick that box, i just unticked it and retried the latest release run... still succeeds. so I'll ammend my pr.

Reply:
I was under the impression you only needed the "Allow GitHub Actions to create and approve pull requests" checkbox to be ticked.

And I thought it was ticked by default for new repos.

The permissions you have outlined here are effectively that check box, no?

Reply:
I guess it does need write since it will create a new branch with the release etc as well. But a per-workflow setting like you have proposed in YAML is explicit more restrictive than allowing Read/Write in "Workflow Permissions" as in that last screenshot.